agent/CMakeLists.txt

```diff
@@ -57,8 +58,14 @@ install(TARGETS softwarecontainer-agent DESTINATION bin)
 # At this point, there should be no more definitions being added/changed so it's OK to set them
 set_all_deferred_config_definitions()
 
+# Network settings should only be in the config if the network gateway is enabled
+if(ENABLE_NETWORKGATEWAY)
+    configure_file(softwarecontainer-config.network.in softwarecontainer-config.network)
```

Comment:
The config configuration (whoa!) has become so non-obvious that it might deserve some comment in the cmake file before configuring things? Like, "the network in-file is configured separately and later made part into the config in-file", or something like that.